cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -67,9 +69,16 @@ type devicePlugin struct {
 
 	scanTicker *time.Ticker
 	scanDone   chan bool
+
+	resMan *rm.ResourceManager
 }
 
 func newDevicePlugin(sysfsDir, devfsDir string, options cliOptions) *devicePlugin {
+	var resourceManager *rm.ResourceManager
+	if options.resourceManagement {
+		resourceManager = rm.NewResourceManager()
+	}
+
```

Comment:
I agree it looks neither smart nor elegant, but adheres better to the principle of least astonishment: in the previous version the name `NewResourceManager()` suggests it is a constructor for `ResourceManager` whereas it may or may not create an actual instance. And a user can't know for sure what is the case without looking into `NewResourceManager()`'s code.

Also you can drop the `var` declaration with
```go
dp := &devicePlugin{
    sysfsDir: sysfsDir,
    ...
    scanDone: make(chan bool, 1),
}

if options.resourceManagement {
    dp.resMan = rm.NewResourceManager()
}

return dp
```